### PR TITLE
Pilin branch

### PR DESCRIPTION
**Pull request by pillmatictm**
Pull request for instructional staff, sorry for late collaboration requests;
MapFrag inflates, but does not display map.

### CODE REVIEW

File: app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapFragment.java [hunk 1]
```diff
@@ -1,63 +1,97 @@
...
+    public void onMapReady(GoogleMap googleMap) {
+        mMap = googleMap;
+        googleMap.setMapType(GoogleMap.MAP_TYPE_NORMAL);
+
+        UiSettings uiSettings = mMap.getUiSettings();
+        uiSettings.setZoomControlsEnabled(true);
+        uiSettings.setMyLocationButtonEnabled(true);
+
+        List<City> cityList = new ArrayList<>();
+        for (City city : cityList) {
```
Comment (mbryantramirez) [inline app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapFragment.java]: is this needed? your only displaying one location in the map fragment not the list of cities

File: app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapFragment.java [hunk 2]
```diff
@@ -1,63 +1,97 @@
...
+        } else {
+            throw new RuntimeException(context.toString()
+                    + " must implement FragmentInterface");
+        }
     }
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         if (getArguments() != null) {
```
Comment (mbryantramirez) [inline app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapFragment.java]: why'd you remove this

File: app/src/main/res/layout/fragment_list.xml [hunk 3]
```diff
@@ -15,8 +15,19 @@
...
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent">
-    </android.support.v7.widget.SearchView>
+        app:layout_constraintTop_toTopOf="parent"/>
+
+    <ProgressBar
+        android:id="@+id/progressBar"
+        android:layout_width="400dp"
+        android:layout_height="120dp"
+        android:layout_marginBottom="8dp"
```
Comment (mbryantramirez) [inline app/src/main/res/layout/fragment_list.xml]: width height and margin should be extracted to the dimens.xml file

File: app/src/main/res/layout/recycler_itemview.xml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    android:layout_height="wrap_content">
+
+    <android.support.v7.widget.CardView
+        android:id="@+id/cityCountry_tile"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="4dp"
+        android:layout_marginTop="4dp"
+        android:layout_marginEnd="4dp"
+        android:layout_marginBottom="4dp"
```
Comment (mbryantramirez) [inline app/src/main/res/layout/recycler_itemview.xml]: dimens.xml

File: app/src/main/res/layout/recycler_itemview.xml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:orientation="horizontal">
+
+
+        <TextView
+            android:id="@+id/cityName"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textSize="30sp"
```
Comment (mbryantramirez) [inline app/src/main/res/layout/recycler_itemview.xml]: dimens.xml

File: app/src/test/java/org/pursuit/chong_pilin_final_assessment/CompareJUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        int result = first.compareTo(second);
+        //actual will be the difference between first and second
+        Assert.assertEquals(result, -8);
+    }
+
+
+    @Test
+    public void test_cityName_testSort() {
+        final City nyc = new City("US", "New York City", 718,null );
+        final City detroit = new City("US", "Detroit", 313, null);
```
Comment (mbryantramirez) [inline app/src/test/java/org/pursuit/chong_pilin_final_assessment/CompareJUnitTest.java]: If you set variables to final you should do so throughout your entire app

File: app/src/test/java/org/pursuit/chong_pilin_final_assessment/CompareJUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class CompareJUnitTest {
+    private City city;
+
+    @Before
+    public void setUp() throws Exception {
+        city = new City(null, null, 0, null);
```
Comment (mbryantramirez) [inline app/src/test/java/org/pursuit/chong_pilin_final_assessment/CompareJUnitTest.java]: city is never used

File: app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/ListFragment.java [hunk 8]
```diff
@@ -1,66 +1,131 @@
...
+                .enqueue(new Callback<List<City>>() {
+                    @Override
+                    public void onResponse(Call<List<City>> call, Response<List<City>> response) {
+                        cityList = response.body();
+                        Collections.sort(cityList);
+                        Log.d(TAG, "onResponse: " + cityList);
+                        countryAdapter = new CountryAdapter(cityList, fragmentInterface);
+                        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+                        recyclerView.setAdapter(countryAdapter);
+                        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
```
Comment (mbryantramirez) [inline app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/ListFragment.java]: try splitting out your search view logic from this method to help make it neater

File: app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapsFragment.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    public static final String TAG = "cityBundle";
+    public static final String NAME = "cityName";
+    public static final String COUNTRY = "country";
+    public static final String LAT = "lat";
+    public static final String LON = "lon";
+    private String cityName;
+    private String country;
+    private double lat;
+    private double lon;
+    private FragmentInterface fragmentInterface;
```
Comment (mbryantramirez) [inline app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapsFragment.java]: these variables are never used are they needed?

File: app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapsFragment.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+    private City city;
+    private City.Coordinates cityCoordinates;
+    private GoogleMap mMap;
```
Comment (mbryantramirez) [inline app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapsFragment.java]: this variable is never used outside of the method onMapReady this doesnt need to be a global variable

File: app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapsFragment.java [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+        }
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            cityName = getArguments().getString(NAME);
+            country = getArguments().getString(COUNTRY);
+            lat = getArguments().getDouble(LAT);
```
Comment (mbryantramirez) [inline app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapsFragment.java]: you get the following error here 
2019-04-23 17:34:59.743 17534-17534/org.pursuit.chong_pilin_final_assessment W/Bundle: Key lon expected Double but value was a java.lang.String.  The default value 0.0 was returned.
2019-04-23 17:34:59.744 17534-17534/org.pursuit.chong_pilin_final_assessment W/Bundle: Attempt to cast generated internal exception:
    java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Double

File: app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapsFragment.java [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+            lon = getArguments().getDouble(LON);
+        }
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        SupportMapFragment mapFragment = (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.googleMap);
```
Comment (mbryantramirez) [inline app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapsFragment.java]: if your setting up a fragment object the only way to get it tot display is through a fragment transaction

File: app/src/main/res/layout/activity_main.xml [hunk 13]
```diff
@@ -4,6 +4,6 @@
     android:id="@+id/fragment_container"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="#59B9FF"
+    android:background="#007BFF"
```
Comment (mbryantramirez) [inline app/src/main/res/layout/activity_main.xml]: color hexcodes should be extracted to colors.xml

File: app/src/main/res/layout/fragment_list.xml [hunk 14]
```diff
@@ -1,32 +1,70 @@
 <?xml version="1.0" encoding="utf-8"?>
 <android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:background="#88FF2D76"
```
Comment (mbryantramirez) [inline app/src/main/res/layout/fragment_list.xml]: colors.xml

File: app/src/main/res/layout/fragment_maps.xml [hunk 15]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/googleMap"
+    android:name="com.google.android.gms.maps.SupportMapFragment"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="#B7FFEB3B"
```
Comment (mbryantramirez) [inline app/src/main/res/layout/fragment_maps.xml]: colors.xml

File: app/src/main/java/org/pursuit/chong_pilin_final_assessment/MainActivity.java [hunk 16]
```diff
@@ -3,11 +3,33 @@
...
+import org.pursuit.chong_pilin_final_assessment.model.City;
+
+public class MainActivity extends AppCompatActivity implements FragmentInterface {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        getSupportFragmentManager()
```
Comment (mbryantramirez) [inline app/src/main/java/org/pursuit/chong_pilin_final_assessment/MainActivity.java]: you should be making your network call for data her and then sorting that data and passing it ListFragment using the new Instance method

File: app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapsFragment.java (hunk 12 above)
Comment (mbryantramirez) [reply app/src/main/java/org/pursuit/chong_pilin_final_assessment/fragments/MapsFragment.java]: what you will need to do is give SujpportMapFragment its own  onMapReadyCallBack listener

**Comment by pillmatictm** [comment]
@mbryantramirez  @JDVila @rusili 
final commit made, merging to master

**Pull request merged**